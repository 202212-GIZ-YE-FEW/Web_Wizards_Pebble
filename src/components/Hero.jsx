import Image from "next/image";
import { Button } from "react-flatifycss";

import heroImage from "~/hero.svg";

function Title(props) {
    const { text } = props;
    return (
        <h1 className='text-black-100 tracking-wide leading-[50px] text-5xl font-medium font-sans lg:h-[88px] lg:w-[500px]'>
            {text}
        </h1>
    );
}

function Description(props) {
    const { text } = props;
    return (
        <p className='text-base text-black-50 leading-7 tracking-[0.1px] lg:w-[500px] lg:h-[90px]'>
            {text}
        </p>
    );
}

function GetStartedButton(props) {
    const { text } = props;
    return (
        <Button className='text-white font-medium bg-[#2F7DA9] shadow-none h-[52px] w-[166px] rounded-lg'>
            {text}
        </Button>
    );
}

export default function Hero(props) {
    const { t } = props;
    return (
        <header className='container flex flex-col-reverse gap-y-5 justify-center items-center my-10 lg:flex-row lg:gap-x-5 lg:gap-y-0'>
            <div className='flex flex-col gap-y-5'>
                <Title text={t("landingTitle")} />
                <Description text={t("landingDescription")} />
                <GetStartedButton text='Get started' />
            </div>
            <Image
                src={heroImage}
                alt='Five people standing together in a farm'
                height={638}
                width={638}
            />
        </header>
    );
}
