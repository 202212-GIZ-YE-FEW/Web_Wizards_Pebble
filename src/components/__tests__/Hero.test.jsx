import { I18nContext } from "next-i18next";
import renderer from "react-test-renderer";

import Hero from "@/components/Hero";

it("renders correctly", () => {
    const i18n = {
        language: "en",
        t: (str) => str,
    };
    const tree = renderer
        .create(
            <I18nContext.Provider value={i18n}>
                <Hero t={i18n.t} />
            </I18nContext.Provider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
