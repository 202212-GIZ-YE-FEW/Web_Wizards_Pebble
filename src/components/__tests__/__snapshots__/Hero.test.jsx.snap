// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="container flex flex-col-reverse gap-y-5 justify-center items-center my-10 lg:flex-row lg:gap-x-5 lg:gap-y-0"
>
  <div
    className="flex flex-col gap-y-5"
  >
    <h1
      className="text-black-100 tracking-wide leading-[50px] text-5xl font-medium font-sans lg:h-[88px] lg:w-[500px]"
    >
      landingTitle
    </h1>
    <p
      className="text-base text-black-50 leading-7 tracking-[0.1px] lg:w-[500px] lg:h-[90px]"
    >
      landingDescription
    </p>
    <button
      className="button__ButtonWrapper-sc-1aw98b4-0 button text-white font-medium bg-[#2F7DA9] shadow-none h-[52px] w-[166px] rounded-lg"
      disabled={false}
    >
      Get started
    </button>
  </div>
  <img
    alt="Five people standing together in a farm"
    data-nimg="1"
    decoding="async"
    height={638}
    loading="lazy"
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg.jpg&w=1920&q=75"
    srcSet="/_next/image?url=%2Fimg.jpg&w=640&q=75 1x, /_next/image?url=%2Fimg.jpg&w=1920&q=75 2x"
    style={
      Object {
        "color": "transparent",
      }
    }
    width={638}
  />
</header>
`;
